src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2160,8 +2160,10 @@ class TableConstraintSegment(BaseSegment):
             "CONSTRAINT", Ref("ObjectReferenceSegment"), optional=True
         ),
         OneOf(
-            Sequence(  # UNIQUE ( column_name [, ... ] )
+            Sequence(  # UNIQUE [INDEX | KEY] [index_name] ( column_name [, ... ] )
                 "UNIQUE",
+                OneOf("INDEX", "KEY", optional=True),
+                Ref("ObjectReferenceSegment", optional=True),
```

Comment:
As far as I can see this is MySQL specific.

Can you copy this segment into the mysql dialect (changing the decorator to `@mysql_dialect.segment(replace=True)`) and remove these changes from ANSI.
i.e.
```sql
@mysql_dialect.segment(replace=True)
class TableConstraintSegment(BaseSegment):
    """A table constraint, e.g. for CREATE TABLE."""
    type = "table_constraint_segment"
    # Later add support for CHECK constraint, others?
    # e.g. CONSTRAINT constraint_1 PRIMARY KEY(column_1)
    match_grammar = Sequence(
        Sequence(  # [ CONSTRAINT <Constraint name> ]
            "CONSTRAINT", Ref("ObjectReferenceSegment"), optional=True
        ),
        OneOf(
            Sequence(  # UNIQUE [INDEX | KEY] [index_name] ( column_name [, ... ] )
                "UNIQUE",
                OneOf("INDEX", "KEY", optional=True),
                Ref("ObjectReferenceSegment", optional=True),
                Ref("BracketedColumnReferenceListGrammar"),
                # Later add support for index_parameters?
            ),
            Sequence(  # PRIMARY KEY ( column_name [, ... ] ) index_parameters
                Ref("PrimaryKeyGrammar"),
                # Columns making up PRIMARY KEY constraint
                Ref("BracketedColumnReferenceListGrammar"),
                # Later add support for index_parameters?
            ),
            Sequence(  # FOREIGN KEY ( column_name [, ... ] )
                # REFERENCES reftable [ ( refcolumn [, ... ] ) ]
                Ref("ForeignKeyGrammar"),
                # Local columns making up FOREIGN KEY constraint
                Ref("BracketedColumnReferenceListGrammar"),
                "REFERENCES",
                Ref("ColumnReferenceSegment"),
                # Foreign columns making up FOREIGN KEY constraint
                Ref("BracketedColumnReferenceListGrammar"),
                # Later add support for [MATCH FULL/PARTIAL/SIMPLE] ?
                AnyNumberOf(
                    # ON DELETE clause, e.g. ON DELETE NO ACTION
                    Sequence(
                        "ON",
                        "DELETE",
                        Ref("TableConstraintReferenceOptionGrammar"),
                    ),
                    # ON UPDATE clause, e.g. ON UPDATE SET NULL
                    Sequence(
                        "ON",
                        "UPDATE",
                        Ref("TableConstraintReferenceOptionGrammar"),
                    ),
                ),
            ),
        ),
    )
```

Also move your unit test to mysql

Reply:
Yeap. Let me check this. At least the MySql documentation says that it was introduced to make MySql compatible to other DBMS. But Postgres is definatly wrong.

It's on my list.